adapters/tlsio_openssl.c

```diff
@@ -390,7 +390,8 @@ static const IO_INTERFACE_DESCRIPTION tlsio_openssl_interface_description =
 };
 
 static LOCK_HANDLE * openssl_locks = NULL;
-
+static int locks_init_count = 0;
+static int openssl_init_count = 0;
```

Comment:
>openssl_init_count [](start = 11, length = 18)

When I reached this line, I realized that my initial statement of the fact that openssl_static_locks_install should not error when multiple calls to init are made is wrong.

I understand the scenario is that the EH SDK is not the only user of openssl and some other users of openssl might initialize openssl also.

But since you are hitting the error in openssl_static_locks_install when openssl_locks is non-NULL, I understand that there are at least 2 calls to tlsio_openssl_init. I think that should not be the case. tlsio_openssl_init is called by platform_init IIRC and that is meant as a one time init for the platform, no nesting supported at the moment.

Nesting can be added, but the question is whether it is needed that platform_init is called multiple times.

If nesting is truly needed, the change looks generally OK. But I'd like to understand the multiple tlsio_openssl_init calls before merging the fix as this area has proven to be tricky in the past.

Reply:
Since there is no coordination between the two openssl clients, and they just happen to both use openssl from the same executable but without any specific mutually agreed timeline or sequence, it appears that only way to allow that to happen is to allow each client to call platform_init and platform_deinit based on their own business logic.